ggshield/cmd/utils/common_options.py

```diff
@@ -76,33 +82,36 @@ def callback(
     return callback
 
 
+def verbose_callback(
+    ctx: click.Context, param: click.Parameter, value: Optional[bool]
+) -> Optional[bool]:
+    if value is not None:
+        ui.ensure_level(ui.Level.VERBOSE)
+    return value
+
+
 _verbose_option = click.option(
     "-v",
     "--verbose",
     is_flag=True,
     default=None,
     help="Verbose display mode.",
-    callback=create_config_callback("verbose"),
+    callback=verbose_callback,
 )
 
 
 def debug_callback(
     ctx: click.Context, param: click.Parameter, value: Optional[bool]
 ) -> Optional[bool]:
     if value is not None:
-        ui.set_level(ui.Level.DEBUG)
         setup_debug_mode()
     return value
 
 
-# The --debug option is marked as "is_eager" so that we can setup logs as soon as
-# possible. If we don't then log commands for the creation of the Config instance
-# are ignored.
```

Comment:
Good question :). It used to be required because the config was loaded by the callback of the `--config` option, so by marking this option as eager its callback was called before the `--config` callback.

It is no longer necessary because the config is now loaded in the body of the `cli()` function, after all option callbacks have been called.